src/mongoc/mongoc-uri.c

```diff
@@ -1314,6 +1315,29 @@ mongoc_uri_set_appname (mongoc_uri_t *uri, const char *value)
    return true;
 }
 
+/* can't use mongoc_uri_get_option_as_int32, it treats 0 specially */
+int32_t
+mongoc_uri_get_local_staleness_option (const mongoc_uri_t *uri)
```

Comment:
Rename "local_threshold_option"? Threshold and staleness are completely unrelated.